module/entities/helper.js

```diff
@@ -1,4 +1,943 @@
+                        if (cloneData.addressLine1) {
+                            delete cloneData.addressLine1;
+                        }
+                        return cloneData;
+                    })
+                }
+
+                let columns = _subEntityListColumns();
+                resolve({
+                    message: constants.apiResponses.ENTITIES_FETCHED,
+                    result: {
+                        count: result.count ? result.count : 0,
+                        columns: columns,
+                        data: result.data
+                    }
+                });
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+    /**
+   * Get either immediate entities or entity traversal based upon the type.
+   * @method
+   * @name subEntities
+   * @param {body} entitiesData
+   * @returns {Array} - List of all immediate entities or traversal data.
+   */
+
+    static subEntities(entitiesData) {
+        return new Promise(async (resolve, reject) => {
+
+            try {
+
+                let entitiesDocument;
+
+                if (entitiesData.type !== "") {
+
+                    entitiesDocument = await this.entityTraversal(
+                        entitiesData.entityId,
+                        entitiesData.type,
+                        entitiesData.search,
+                        entitiesData.limit,
+                        entitiesData.pageNo
+                    );
+                } else {
+
+                    entitiesDocument = await this.immediateEntities(
+                        entitiesData.entityId,
+                        entitiesData.search,
+                        entitiesData.limit,
+                        entitiesData.pageNo
+                    );
+                }
+
+
+                return resolve(entitiesDocument);
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+    /**
+   * Get immediate entities.
+   * @method
+   * @name entityTraversal
+   * @param {Object} entityId
+   * @returns {Array} - List of all immediateEntities based on entityId.
+   */
+
+    static entityTraversal(
+        entityId,
+        entityTraversalType = "",
+        searchText = "",
+        pageSize,
+        pageNo
+    ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let entityTraversal = `groups.${entityTraversalType}`;
+
+                let entityDocs =
+                    await this.entityDocuments(
+                        {
+                            _id: entityId,
+                            "groups": { $exists: true },
+                            [entityTraversal]: { $exists: true }
+                        },
+                        [entityTraversal]
+                    );
+
+                let entitiesDocument = entityDocs.data;
+
+                if (!entitiesDocument[0]) {
+                    return resolve([]);
```

Comment:
@rakeshSgr - Lets not resolve blank objects, every helper function can return as below format

{
success : true/false // true by default, false if returned from catch
message : "" // some text explaining
data : false / array / object // by default false
}